File: src/main/java/com/nicomincuzzi/frameworkless/db/PostgreSQLJDBC.java
package com.nicomincuzzi.frameworkless.db;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class PostgreSQLJDBC implements JDBCDriver {

    @Override
    public Connection connect() throws Exception {
        String url = "jdbc:postgresql://localhost/test";
        Properties props = new Properties();
        props.setProperty("user", "fred");
        props.setProperty("password", "secret");
        props.setProperty("ssl", "true");
        try {
            return DriverManager.getConnection(url, props);
        } catch (SQLException e) {
            throw new Exception();
        }
    }
}

File: src/main/kotlin/com/nicomincuzzi/frameworkless/infrastructure/db/PostgreSQLJDBC.kt
package com.nicomincuzzi.frameworkless.infrastructure.db

import java.lang.Exception
import java.sql.Connection
import java.util.Properties
import java.sql.DriverManager
import java.sql.SQLException

class PostgreSQLJDBC : JDBCDriver {
    @Throws(Exception::class)
    override fun connect(): Connection? {
        val url = "jdbc:postgresql://localhost/test"
        val props = Properties()
        props.setProperty("user", "fred")
        props.setProperty("password", "secret")
        props.setProperty("ssl", "true")
        return try {
            DriverManager.getConnection(url, props)
        } catch (e: SQLException) {
            throw Exception()
        }
    }
}